### PR TITLE
Update existing and introduce new resources/datasources into the upcloud Terraform provider

### PR DESCRIPTION
**Pull request by trentrosenbaum**
This PR represents the implementation of new resources and datasources that allow the UpCloud Terraform provider to operate with an UpCloud account through the existing UpCloud 1.3 API.  The PR updates a subset of the existing resources and removes resources where they do not fit into the Terraform workflow.  Each of the resource and datasources are delivered with expanded acceptance tests and website documentation.

The following is a list of alterations documented in the changelog.  The terraform provider has been updated to use the `Terraform Plugin SDK V2` repo and the readme has been expanded to direct developers for local development with Terraform 0.13.  New makefile targets have been added to allow website docs to be viewed locally and the provider now depends on Go 1.14 and go modules.  For building through pipelines the vendor directory is still active.

### Added
 - datasource_upcloud_hosts to view hosts data
 - datasource_upcloud_ip_addresses to retrieve account wide ip address data
 - datasource_upcloud_networks to retrieve account wide networks data
 - datasource_upcloud_tags to retrieve account wide tag data
 - datasource_upcloud_zone to retrieve specific zone details
 - datasource_upcloud_zones to retrieve account wide zone data
 - resource_upcloud_firewall_rules add to allow rules to be applied to server
 - resource_upcloyd_floating_ip_address to allow the management of floating ip addresses
 - resource_upcloud_network to allow the management of networks
 - resource_upcloud_router to allow the management of routers

### Changed
 - resource_upcloud_server expanded with new functionality from UpCloud API 1.3
 - resource_upcloud_storage expaned with new functionalty from UpCloud API 1.3
 - resource_upclopud_tag expanded to implement read function
 
### Removed
 - resource_upcloud_ip removed and replaced by resource_upcloud_floating_ip_address
 - resource_upcloud_firewall_rule removed and replaced by resource_upcloud_firewall_rules
 - resource_upcloud_zone removed and replaced by zone and zones datasources

### CODE REVIEW

File: upcloud/resource_upcloud_server_test.go [hunk 1]
```diff
@@ -65,6 +78,204 @@ func TestUpcloudServer_changePlan(t *testing.T) {
...
+		return nil
+	}
+}
+
+func testAccCheckNetwork(resourceName string, niIdx int, networkResourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		server := s.RootModule().Resources[resourceName]
+		network := s.RootModule().Resources[networkResourceName]
+		if network == nil {
+			fmt.Printf("%#v", s.RootModule().Resources)
```
Comment (PopoSensei) [inline upcloud/resource_upcloud_server_test.go:251]: Is this a leftover or should we print at the other error states as well?

Comment (PopoSensei) [reply upcloud/resource_upcloud_server_test.go:251]: Or maybe there is another reason for this?

**Review by PopoSensei — CHANGES_REQUESTED** [review]

**Comment by PopoSensei** [comment]
Other than the small things listed before didn't find anything. So LGTM!

File: upcloud/resource_upcloud_server_test.go (hunk 1 above)
Comment (yoink00) [reply upcloud/resource_upcloud_server_test.go]: It's a bit of debug logging. The error returned should be enough to capture the test failure.

**Review by PopoSensei — APPROVED** [review]

**Pull request merged**